### PR TITLE
Changes to Forms doc

### PR DESCRIPTION
**Pull request by ericnakagawa**
Fixes [8052](https://github.com/facebook/react/issues/8052)
Fixes [8068](https://github.com/facebook/react/issues/8068)

### CODE REVIEW

File: docs/docs/forms.md [hunk 1]
```diff
@@ -2,11 +2,12 @@
 id: forms
 title: Forms
 permalink: docs/forms.html
-prev: state-and-lifecycle.html
-next: lifting-state-up.html
-redirect_from: "/tips/controlled-input-null-value.html"
+prev:
```
Comment (gaearon) [inline docs/docs/forms.md]: Why were these removed? Every page in the "Basic Guides" has this links.

Compare Lists and Keys (has these links):

screen shot 2016-10-25 at 12 06 15

and Forms after this change (doesn't have these links anymore, why?):

screen shot 2016-10-25 at 12 06 52

File: docs/docs/forms.md [hunk 2]
```diff
@@ -2,11 +2,12 @@
...
 title: Forms
 permalink: docs/forms.html
-prev: state-and-lifecycle.html
-next: lifting-state-up.html
-redirect_from: "/tips/controlled-input-null-value.html"
+prev: 
+next: 
 ---
 
+## Forms
```
Comment (gaearon) [inline docs/docs/forms.md]: I intentionally removed this subheader in the past.
If you add it, this is how the page looks:

screen shot 2016-10-25 at 12 07 23

The double heading is unnecessary. If you look at other pages, none of the guides have double headings.

File: docs/docs/forms.md [hunk 3]
```diff
@@ -61,7 +62,7 @@ class Form extends React.Component {
 ReactDOM.render(<Form />, document.getElementById('root'));
 ```
 
-[Try it on CodePen.](https://codepen.io/ericnakagawa/pen/QKkJva?editors=0010)
+[Try it on Codepen](https://codepen.io/ericnakagawa/pen/QKkJva?editors=0010)
```
Comment (gaearon) [inline docs/docs/forms.md]: I intentionally changed it to "CodePen" the last time because [this is how the service is capitalized](https://twitter.com/codepen):

screen shot 2016-10-25 at 12 08 33

What is the reason for changing it back?

File: docs/docs/forms.md [hunk 4]
```diff
@@ -114,8 +115,6 @@ class Form extends React.Component {
     return (
       <div>
         <input type="text"
-          placeholder="Hello!"
```
Comment (gaearon) [inline docs/docs/forms.md]: Why do we remove the placeholder too? It is not related to controlled-ness of the components.

File: docs/docs/forms.md [hunk 5]
```diff
@@ -126,7 +125,7 @@ class Form extends React.Component {
 ReactDOM.render(<Form />, document.getElementById('root'));
 ```
 
-[Try it on CodePen.](https://codepen.io/ericnakagawa/pen/XjxyoL?editors=0010)
+[Try it on Codepen](https://codepen.io/ericnakagawa/pen/XjxyoL?editors=0010)
```
Comment (gaearon) [inline docs/docs/forms.md]: Same here: I intentionally changed it to CodePen before.

File: docs/docs/forms.md [hunk 6]
```diff
@@ -259,10 +258,10 @@ class Form extends React.Component {
 ReactDOM.render(<Form />, document.getElementById('root'));
 ```
 
-[Try this on CodePen.](https://codepen.io/ericnakagawa/pen/JRmZjz?editors=0010)
+[Try this on Codepen.](https://codepen.io/ericnakagawa/pen/JRmZjz?editors=0010)
```
Comment (gaearon) [inline docs/docs/forms.md]: Same here: I intentionally changed it to CodePen before.

File: docs/docs/forms.md [hunk 7]
```diff
@@ -298,9 +297,9 @@ class Form extends React.Component {
 ReactDOM.render(<Form />, document.getElementById('root'));
 ```
 
-[Try this on CodePen.](https://codepen.io/ericnakagawa/pen/kkAQPp?editors=0010)
+[Try this on Codepen.](https://codepen.io/ericnakagawa/pen/kkAQPp?editors=0010)
```
Comment (gaearon) [inline docs/docs/forms.md]: Same here: I intentionally changed it to CodePen before.

File: docs/docs/forms.md [hunk 8]
```diff
@@ -336,9 +335,9 @@ class Form extends React.Component {
 ReactDOM.render(<Form />, document.getElementById('root'));
 ```
 
-[Try this on CodePen.](https://codepen.io/ericnakagawa/pen/pExQbL?editors=0010)
+[Try this on Codepen.](https://codepen.io/ericnakagawa/pen/pExQbL?editors=0010)
```
Comment (gaearon) [inline docs/docs/forms.md]: Same here: I intentionally changed it to CodePen before.

File: docs/docs/forms.md [hunk 9]
```diff
@@ -417,7 +421,7 @@ class Form extends React.Component {
 ReactDOM.render(<Form />, document.getElementById('root'));
 ```
 
-[Try it on CodePen.](https://codepen.io/ericnakagawa/pen/kkAzPO?editors=0010)
+[Try it on Codepen](https://codepen.io/ericnakagawa/pen/kkAzPO?editors=0010)
```
Comment (gaearon) [inline docs/docs/forms.md]: Same here: I intentionally changed it to CodePen before.

File: docs/docs/forms.md [hunk 10]
```diff
@@ -389,7 +390,7 @@ class Form extends React.Component {
 
   handleChange(event) {
     let val = event.target.value;
```
Comment (gaearon) [inline docs/docs/forms.md]: Both of these can be `const`, not `let`.

File: docs/docs/forms.md [hunk 11]
```diff
@@ -389,7 +390,7 @@ class Form extends React.Component {
...
-    let checked = this.state.checked.slice(); // copy
+    let checked = this.state.checked.slice(); // copy 
     if(checked.includes(val)) {
```
Comment (gaearon) [inline docs/docs/forms.md]: Please put space between `if` and the opening brace.
Please don't use `includes()`, it's a ES2016 API that is unsupported by any other than most recent browsers.

File: docs/docs/forms.md [hunk 12]
```diff
@@ -114,8 +115,6 @@ class Form extends React.Component {
     return (
       <div>
         <input type="text"
-          placeholder="Hello!"
-          value={this.state.value}
```
Comment (gaearon) [inline docs/docs/forms.md]: Can you please add line highlighting for the relevant lines in these examples? They're pretty long, and it would help to highlight whatever parts or properties that are being explained.

File: docs/docs/forms.md [hunk 13]
```diff
@@ -298,9 +297,9 @@ class Form extends React.Component {
 ReactDOM.render(<Form />, document.getElementById('root'));
 ```
 
-[Try this on CodePen.](https://codepen.io/ericnakagawa/pen/kkAQPp?editors=0010)
+[Try this on Codepen.](https://codepen.io/ericnakagawa/pen/kkAQPp?editors=0010)
 
-### Basic Select
+### Basic Uncontrolled Select
```
Comment (gaearon) [inline docs/docs/forms.md]: Isn't it controlled? The example is setting `value` on `select`.

File: docs/docs/forms.md [hunk 14]
```diff
@@ -360,9 +359,12 @@ class Form extends React.Component {
   render() {
     return (
       <div>
-        <input type="radio" name="choice" value="A" onChange={this.handleChange} /> Option A<br />
-        <input type="radio" name="choice" value="B" onChange={this.handleChange} defaultChecked={true} /> Option B<br />
-        <input type="radio" name="choice" value="C" onChange={this.handleChange} /> Option C<br />
+        <input type="radio" name="choice" value="A"
```
Comment (gaearon) [inline docs/docs/forms.md]: This is not very readable. There is also a problem: clicking on the label doesn't select the option. This is not a very good UX.

I also don't think `name` attribute is relevant in this example.

We should use `<label>` instead, like this:

```js
        <label>
          <input type="radio" value="A" onChange={this.handleChange} />
          Option A
        </label>
```

**Comment by gaearon** [comment]
On a broader scale, I'm not sure why we should controlled examples for input, textarea, and select, but uncontrolled radio and checkbox. Can we show a controlled and uncontrolled example for each of them?

I also feel that this document is too long and has too many details compared to other "basics" guide. If I was learning React I would stop reading somewhere in the middle, and miss the next important topics.

Can we split this guide into "Controlling Inputs" and "Forms in Depth" or something like this?

Also, can you please move the events section to the events reference as we discussed?

**Comment by lacker** [comment]
Did something weird happen while merging? This pull request looks like it is undoing a bunch of stuff that it doesn't seem like you intended to undo. Also you should target the facebook:master branch rather than facebook:new-docs now that new-docs is merged.

**Comment by ericnakagawa** [comment]
Applied changes in this PR: https://github.com/facebook/react/pull/8112

**Comment by ericnakagawa** [comment]
@gaearon I like the idea of lightening up this document. I think it would be good to break this out in a future pass.
